Toshi/Source/Toshi/File/TCompress_Decompress.h

```diff
@@ -17,7 +17,7 @@ namespace Toshi
 		};
 	
 		static uintptr_t Decompress(TFile* file, Header* header, void* buffer, uint32_t& size);
-		static int8_t ReadHeader(TFile* file, Header& btecHeader);
-		static int GetReadCount(TFile* file, uint32_t& read_dst, uint32_t& size, int& offset);
+		static int8_t ReadHeader(TFile* file, Header& btecHeader, uint16_t& headerSize);
```

Comment:
Are there 3 parameters in the orginial func too?

Reply:
no, the third parameter is not in the original function

Reply:
Hm, should we then do it like i did before or your approach? We could make it like the original TOSHI engine did, if we could decompile the func where BTEC 1.3 was used

Reply:
IIRC it was in nicktoons battle for volcano island

Reply:
I added the third parameter because it helps to track the size of header, it's needed in TTSF::DecompressSection. the original Toshi doesn't support few versions of the compression algorithms at the same time (at least in De Blob, if I'm not mistaken)

Reply:
> Hm, should we then do it like i did before or your approach? We could make it like the original TOSHI engine did, if we could decompile the func where BTEC 1.3 was used

the problem was in the TTSF::DecompressSection function, m_ReadPos had been been always increased by sizeof(Header) even when there was no XOR value

Reply:
> IIRC it was in nicktoons battle for volcano island

which edition should I get, PS2 or Gamecube?

Reply:
Ps2 is little endian
Gamecube is big endian
Idk i have ps2 but i think gamecube is better to reverse

Reply:
I removed the third parameter in the latest commit